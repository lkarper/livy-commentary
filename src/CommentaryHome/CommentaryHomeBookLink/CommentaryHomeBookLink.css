.CommentaryHomeBookLink__a {
    color: black;
    text-decoration: none;
}

.CommentaryHomeBookLink__a:hover, .CommentaryHomeBookLink__a:focus {
    text-decoration: underline;
}