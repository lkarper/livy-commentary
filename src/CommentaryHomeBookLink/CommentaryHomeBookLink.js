import React from 'react';
import { Link } from 'react-router-dom';
import CommentaryHomeChapterLink from '../CommentaryHomeChapterLink/CommentaryHomeChapterLink';

const CommentaryHomeBookLink = (props) => {

    const { book } = props;

    return (
        <div>
            <Link>Book {book.book_number}</Link>
            {book.chapters.map(chapter => <CommentaryHomeChapterLink key={chapter.chapter_number} chapter={chapter} />)}
        </div>
    );
}

export default CommentaryHomeBookLink;