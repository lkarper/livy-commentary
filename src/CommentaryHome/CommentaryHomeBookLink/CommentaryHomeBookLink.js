import React from 'react';
import { Link } from 'react-router-dom';
import CommentaryHomeChapterLink from '../CommentaryHomeChapterLink/CommentaryHomeChapterLink';
import './CommentaryHomeBookLink.css';

const CommentaryHomeBookLink = (props) => {

    const { book } = props;

    return (
        <div className='CommentaryHomeBookLink_book-link-container'>
            <h2>
                <Link 
                    to={`/commentary-read?book=${book.book_number}`}
                >
                    Book {book.book_number}
                </Link>
            </h2>
            {book.chapters.map(chapter => 
                <CommentaryHomeChapterLink 
                    key={chapter.chapter_number} 
                    book_number={book.book_number} 
                    chapter={chapter} 
                />)
            }
        </div>
    );
}

export default CommentaryHomeBookLink;