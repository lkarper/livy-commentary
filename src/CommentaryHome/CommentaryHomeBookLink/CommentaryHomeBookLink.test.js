import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import CommentaryHomeBookLink from './CommentaryHomeBookLink';

describe('CommentaryHomeBookLink component', () => {
    it('renders the UI as expected', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <CommentaryHomeBookLink />
            </BrowserRouter>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(
            <BrowserRouter>
                <CommentaryHomeBookLink />
            </BrowserRouter>
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
