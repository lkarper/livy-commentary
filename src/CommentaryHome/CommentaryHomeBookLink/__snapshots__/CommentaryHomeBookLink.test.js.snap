// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentaryHomeBookLink component renders the UI as expected 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <CommentaryHomeBookLink
      book={
        Object {
          "book_number": 0,
          "chapters": Array [],
        }
      }
    >
      <div
        className="CommentaryHomeBookLink_book-link-container"
      >
        <h2>
          Error
        </h2>
        <p>
          Looks like something went wrong. Please check your connection and try again.
        </p>
      </div>
    </CommentaryHomeBookLink>
  </Router>
</BrowserRouter>
`;
